### PR TITLE
Categorise the various compilers by licence.

### PR DESCRIPTION
**Pull request by Lucretia**
Organised the compilers into closed/open source and by which licence if any. This might clear up some of the issues new users have regarding GNAT.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -180,16 +181,28 @@ A curated list of awesome resources related to the Ada and SPARK programming lan
 
 ## Compilers
 
-#### Local
-- [gnat-ce](https://www.adacore.com/download) - Adacore compiler for the Ada programming language which forms part of the GNU Compiler Collection. It supports all versions of the language, i.e. Ada 2012, Ada 2005, Ada 95 and Ada 83.
-- [fsf-gnat](http://www.getadanow.com/) - Free Software Foundation compiler for the Ada programming language which forms part of the GNU Compiler Collection. It supports all versions of the language, i.e. Ada 2012, Ada 2005, Ada 95 and Ada 83.
+#### Commercial / Closed source
```
Comment (Fabien-Chouteau) [inline README.md]: Commercial and closed source are two different things. I don't see why they should be grouped under the same category.

Also GNAT pro is missing from the list and it is a good example of commercial and open-source.

Comment (Lucretia) [reply README.md]: Commercial generally means closed source and those compilers listed under that category are both.

As for GNAT Pro, then maybe there should be a Commercial / Open source category added? The fact that GNAT Pro wasn't there, is the reason that there is no category for it.

File: README.md [hunk 2]
```diff
@@ -180,16 +181,28 @@ A curated list of awesome resources related to the Ada and SPARK programming lan
...
 - [greenhills-ada](https://www.ghs.com/products/ada_optimizing_compilers.html) - Green Hills Software offers the industry's first, fully validated family of Ada Cross Compilers for real-time targets.
+- [xgc](http://www.xgc.com/) - XGC provide GCC-based bare-metal toolchains with Ada95 support for aerospace-specific targets such as MIL1750, ERC32 and LEON. As well as a legacy M68000 toolchain.
+
+#### Open source
+
+##### GPL (no linking exception)
+
+- [gnat-ce](https://www.adacore.com/download) - Adacore compiler for the Ada programming language which forms part of the GNU Compiler Collection. It supports all versions of the language, i.e. Ada 2012, Ada 2005, Ada 95 and Ada 83.
+
+##### GPL (with linking exception)
```
Comment (ohenley) [inline README.md]: fsm-gnat has 'linking exception' or is it the other way around? Sorry not much in the 'know' about compilers.

Comment (Lucretia) [reply README.md]: FSF GNAT has the linking exception applied. Like I said, I've done this to help newcomers to the language understand the compilers, specifically gnat, as that is what they'll use.

File: README.md (hunk 1 above)
Comment (ohenley) [reply README.md]: @Fabien-Chouteau Can you edit to add the proper GNAT pro and decide which are commercial closed and commercial opened?

File: README.md (hunk 2 above)
Comment (ohenley) [reply README.md]: ok. Thx for the precision. linking exception to what, for example?

File: README.md (hunk 1 above)
Comment (Lucretia) [reply README.md]: Already done.

Comment (ohenley) [reply README.md]: @Fabien-Chouteau I will merge this one. If/when you have the time, please make a pr adding GNAT-Pro and a more detailed caterogy for Commercial, close and Commercial, open.
Thx!

Comment (Lucretia) [reply README.md]: Maybe that is better, uses bold in parentheses.

File: README.md (hunk 2 above)
Comment (Lucretia) [reply README.md]: As in linking objects against the runtime.

File: README.md [hunk 3]
```diff
@@ -180,16 +182,32 @@ A curated list of awesome resources related to the Ada and SPARK programming lan
...
+
+#### Commercial / Open source
+
+- [GNAT Pro](https://www.adacore.com/)
+
+#### Open source
+
+##### GPL (no linking exception)
+
+- [gnat-ce](https://www.adacore.com/download) - Adacore compiler for the Ada programming language which forms part of the GNU Compiler Collection. It supports all versions of the language, i.e. Ada 2012, Ada 2005, Ada 95 and Ada 83.
```
Comment (Fabien-Chouteau) [inline README.md]: Sorry I didn't catch that before, but GNAT Community supports only Ada 2012.

**Comment by Lucretia** [comment]
Hi, you merged too early, I just pushed some other commits to this branch to merge.

File: README.md (hunk 3 above)
Comment (Fabien-Chouteau) [reply README.md]: See #39.

**Pull request merged**